### ISSUE TITLE
Tips

### ISSUE THREAD

**Issue by simonv3** [openfarmcc/OpenFarm#847]
Provide a way to add gardening tips to OpenFarm.

Tips can be freestanding or linked to other objects (eg. Guides or Crops) and have an optional location associated with it.

Tips can range from what to do about specific pests and how to mulch a garden plot.

Eg. 

Tip title: Mulch your garden
Tip content: Lorem Ipsum
Tip images: [ ] [ ] 
Tip location: Skagway, AK
Tip links: guide#231, guide#43

It would be cool to have tips on the front page. 

Further question: can tips be searched? (I would say yes, but how do we start doing that?)

**Comment by CloCkWeRX** [comment]
Would you go so far as to add in external guides as a link-based-tip? I found that I ended up googling frequently when writing guides for the specific details: I know what I know from online resources - kind of like Wikipedia's reference system.

**Comment by simonv3** [comment]
@CloCkWeRX I'm not sure I fully get what you mean - are you thinking just a tip that is a link to an external source (ex. a wikihow guide on growing garlic)? I think that'd be cool as it would mean less content writing on our side and being able to pull in things fairly quickly. But architecturally I'm not sure what that would look like alongside this (input welcome!).

**Comment by CloCkWeRX** [comment]
>  tip that is a link to an external source (ex. a wikihow guide on growing garlic)
Pretty much. The problem is it moves away from the open licensed data model a little.

Re data model: Crop has many ExternalGuides(title, href); similar to the images modelling. Would suggest keeping it limited to Guides only for the short term; otherwise you get into a murky "external resources(href, title, type)" modelling and before you know it you've got RDF...

**Comment by roryaronson** [comment]
I like this! It sounds like Tips are basically Guides though, just not necessarily linked to a Crop. So would it make sense to make Guides more broad by changing the architecture from:
* Every Guide is linked to one Crop

To:
 * Every Guide is freestanding by default. Guides can optionally be linked to any of the following:
   * Crops
   * Other Guides
   * External Resources

In this way the Guide model covers lots of things:

- Growing Guides for a single Crop
- Growing Guides for a set of Crops (how to take care of the [Three Sisters](https://en.wikipedia.org/wiki/Three_Sisters_(agriculture)) for example)
- Growing Guides/Tips that apply more generally to many Crops (how to prune a fruit tree)
- Tips that don't have to do with any Crop in particular or any Crop at all (freestanding, but could maybe be linked to other Tip-y content or external resources)
- Content that is written somewhere else (we're basically just a directory/search engine for it)

**Comment by roryaronson** [comment]
The main advantage I see in going about this in the way mentioned above is we can keep/expand upon the Guide Creation Flow and inline editing rather than making/maintaining something entirely new that is 90% similar. It also makes OpenFarm easier to understand I think: Its a place for Guides. Some Guides are about how to grow a crop, some are about how to do canning, some are just links to other places with good info.

It would be interesting to see people "hack" the software to their needs/edge cases. For example if there were a bunch of Guides/external resources for canning and preservation, I would consider making a Garden called "Pantry" and then save all of my favorite Guides to that Garden. If there were enough Guides for cooking/preparation, I could have a Garden named "Kitchen" and save all those things there. Its definitely pushing the boundaries of what stuff is designed/intended for, but I bet people will do interesting things like that.

**Comment by roryaronson** [comment]
Also, we've talked about Guides linked to multiple Crops before in #715

**Comment by simonv3** [comment]
I've totally thought about making guides for sourdough, and kombucha, for example.

**Comment by sophiakc** [comment]
Yes indeed, growing guides are the core value of OF: and author could choose to add "snippet" to the guide page, to enhance their story: how to prepare the soil in a specific region, crop definition, forum question they find relevant...

How a visitor could interact / add content to an existing guide?

**Comment by simonv3** [comment]
That's definitely a UI feature we'd have to think more about.
